src/jobserver_test.cc

```diff
@@ -0,0 +1,357 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "jobserver.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#include <cassert>
+#include <cstdlib>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+/// Wrapper class to provide access to protected members of the Jobserver class.
+struct JobserverTest : public Jobserver {
```

Comment:
nit: this class is not really a test, and a name like `TestJobserver` or `JobserverForTest` might be more appropriate. This would allow you to rename the class below from `JobserverTestState` to `JobserverTest`, which is consistent with the way all other unit-tests name sub-classes for `::testing::Test` in Ninja (and many other projects).

Reply:
bikeshedding here: what about `struct JobserverWrapper : public Jobserver {` ?

Reply:
That would work too :-)

Reply:
Wrapper it is then. Thanks.